packages/flutter/lib/src/rendering/grid.dart

```diff
@@ -73,65 +273,78 @@ class RenderGrid extends RenderBox with ContainerRenderObjectMixin<RenderBox, Gr
       child.parentData = new GridParentData();
   }
 
+  Size _getIntrinsicSize(BoxConstraints constraints) {
+    return constraints.constrain(_delegate.getGridSpecification(constraints, childCount).gridSize);
+  }
+
   double getMinIntrinsicWidth(BoxConstraints constraints) {
     assert(constraints.isNormalized);
-    // We can render at any width.
-    return constraints.constrainWidth(0.0);
+    return _getIntrinsicSize(constraints).width;
   }
 
   double getMaxIntrinsicWidth(BoxConstraints constraints) {
     assert(constraints.isNormalized);
-    double maxWidth = childCount * _maxChildExtent;
-    return constraints.constrainWidth(maxWidth);
+    return _getIntrinsicSize(constraints).width;
```

Comment:
It's not clear there's a meaningful answer to this question for FixedColumnCountGridDelegate.  As the FixedColumnCountGrid gets wider, it also gets taller because each tile is a fixed aspect ratio...